### PR TITLE
Add settings option for Evidence double-click that can be turned off (single-click evidence view) and more

### PR DESCRIPTION
**Pull request by Crystalwarrior**
Add a settings option for evidence double click that is on by default, disabling it allows you to open evidence to view in a single click.

Fix name not being set to read only when evidence is initialized
If evidence is double-clicked and overlay is visible, trigger "on_x_clicked" behavior
Make sure evidence is selected when double-click is called
Stop hovering from changing name text if overlay is visible
Simplify on_evidence_x_clicked "modified" check by simply checking if "ui_evidence_ok" is visible

This PR also ends up improving evidence to allow theme layouts where overlay is not forced to block evidence list

### CODE REVIEW

**Review by oldmud0 — APPROVED** [review]
Add to courtroom design docs

**Comment by Salanto** [comment]
Ah yes, the double click interprets as a single click design element. Why is this a theme option, CW?
This would be better suited in the settings menu to work across multiple themes.

**Review by in1tiate — DISMISSED** [review]
please do not add client settings to the theme. i know it's painful to work with AOOptionsDialog, but this evidence one-click thing needs to be user-configurable, there's no reason the theme should be controlling it

File: src/evidence.cpp [hunk 1]
```diff
@@ -106,6 +107,10 @@ void Courtroom::initialize_evidence()
 
 void Courtroom::refresh_evidence()
 {
+  // If only a single click is needed to view evidence, doubleclick by default
+  QString one_click_view = ao_app->get_design_element("evidence_one_click_view", "courtroom_design.ini");
```
Comment (TrickyLeifa) [inline src/evidence.cpp:111]: How are we supposed to test this without a theme sample?

**Review by TrickyLeifa — DISMISSED** [review]

**Comment by Crystalwarrior** [comment]
> please do not add client settings to the theme. i know it's painful to work with AOOptionsDialog, but this evidence one-click thing needs to be user-configurable, there's no reason the theme should be controlling it

I fully agree, will change the PR to be a settings option instead

**Review by oldmud0 — APPROVED** [review]
Anything to fix evidence again

**Pull request merged**